spec/index.spec.js

```diff
@@ -615,6 +615,54 @@ describe('server', () => {
     expect(config.masterKeyCache.expiresAt.getTime()).toBeGreaterThan(Date.now());
   });
 
+  it('should load publicServerURL', async () => {
+    await reconfigureServer({
+      publicServerURL: () => 'https://myserver.com/1',
+    });
+
+    await new Parse.Object('TestObject').save();
+
+    const config = Config.get(Parse.applicationId);
+    expect(config.publicServerURL).toEqual('https://myserver.com/1');
+  });
+
+  it('should load publicServerURL from Promise', async () => {
+    await reconfigureServer({
+      publicServerURL: () => Promise.resolve('https://async-server.com/1'),
+    });
+
+    await new Parse.Object('TestObject').save();
+
+    const config = Config.get(Parse.applicationId);
+    expect(config.publicServerURL).toEqual('https://async-server.com/1');
+  });
+
+  it('should handle publicServerURL function throwing error', async () => {
+    const errorMessage = 'Failed to get public server URL';
+    await reconfigureServer({
+      publicServerURL: () => {
+        throw new Error(errorMessage);
+      },
+    });
+
+    // The error should occur when trying to save an object (which triggers loadKeys in middleware)
+    await expectAsync(
+      new Parse.Object('TestObject').save()
+    ).toBeRejected();
+  });
+
+  it('should handle publicServerURL Promise rejection', async () => {
+    const errorMessage = 'Async fetch of public server URL failed';
+    await reconfigureServer({
+      publicServerURL: () => Promise.reject(new Error(errorMessage)),
+    });
+
+    // The error should occur when trying to save an object (which triggers loadKeys in middleware)
+    await expectAsync(
+      new Parse.Object('TestObject').save()
+    ).toBeRejected();
+  });
```

Comment:
@coderabbitai make a review suggestion to add a test that verifies that every time `publicServerURL` is called, the function is executed and a different value is returned. This prevents future bugs in which the function is only executed once and then a static value is returned on future retrievals of `publicServerURL`. Use for example a counter that returns a value +1 on every retrieval.